src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -2220,7 +2238,8 @@ private void getFieldInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_MET
 
             pResult.fieldAccessor = fieldAccessor;
             pResult.fieldFlags = fieldFlags;
-            pResult.fieldType = getFieldType(pResolvedToken.hField, ref pResult.structType, pResolvedToken.hClass);
+            fixed (CORINFO_CLASS_STRUCT_** pStructType = &pResult.structType)
```

Comment:
Maybe passing in address of the local would look better?

```
CORINFO_CLASS_STRUCT_ structType;
pResult.fieldType = getFieldType(pResolvedToken.hField, &structType, pResolvedToken.hClass);
pResult.structType = structType;
```

Reply:
Or change `pResult` to be a pointer too. We have mix of refs and pointers for similar arguments in this file, and leaning towards pointers overtime to deal with problems like this.

Reply:
> Or change `pResult` to be a pointer too. We have mix of refs and pointers for similar arguments in this file, and leaning towards pointers overtime to deal with problems like this.

True. I ended up putting a couple other places on the pointer plan to avoid having the duplicate implementation of `asCorInfoType`, but seems like we ended up with a change that doesn't make the codebase worse, so that's definitely a plus.

It probably won't merge clean in the r2r branch anymore, but the changes should be obvious.